### DISCUSSION TITLE
Cannot revalidate firestore auto id  in ISR (Next 12)

### DISCUSSION THREAD

**Discussion by thanhtutzaw** [vercel/next.js#42512] [Help]
I use next-firebase-auth library and when I fetch data with SSR It needs browser refresh to Update data . So I added **ISR** and It says 
`Error: Additional keys were returned from `getServerSideProps`. Properties intended for your component must be nested
under the `props` key, e.g.:return { props: { title: 'My Title', content: '...' } }Keys that need to be moved: revalidate.`

my ISR

![carbon](https://user-images.githubusercontent.com/71011043/200096735-284de480-3433-4998-ab88-d13c7d8b6777.png)

**Comment by dev-afzalansari** [comment]
@thanhtutzaw

Use [getStaticProps](https://nextjs.org/docs/basic-features/data-fetching/incremental-static-regeneration) to acheive **ISR** in your application.
